proxy/SessionPoolInterface.h

```diff
@@ -0,0 +1,178 @@
+/** @file
+
+  ProxySession - Base class for protocol client sessions.
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#pragma once
+
+#include "ProxySession.h"
+
+class SessionPoolInterface : public ProxySession
```

Comment:
I'd name this `PoolableProxySession` if we extend `ProxySession`.

If we want to have an *interface*, I'd do
```
class Http1ServerSession : public ProxySession, public SessionPoolInterface
```
or
```
class PoolableProxySession : public ProxySession, public SessionPoolInterface
class Http1ServerSession : public PoolableProxySession
```
```